azure/defaults.go

```diff
@@ -329,6 +335,61 @@ func UserAgent() string {
 	return fmt.Sprintf("cluster-api-provider-azure/%s", version.Get().String())
 }
 
+// ARMClientOptions returns default ARM client options for CAPZ SDK v2 requests.
+func ARMClientOptions(resourceManagerEndpoint string) *arm.ClientOptions {
+	opts := &arm.ClientOptions{}
+	switch resourceManagerEndpoint {
+	case "https://management.azure.com/":
```

Comment:
I couldn't find any in the SDK. I can create constants in CAPZ if we want.

Reply:
looking at this again, if we're just trying to get the cloud name, I think we should be able to use `AzureCluster.Spec.AzureEnvironment` (we'd need to pass it through just like subscriptionID)

I think that would be more resilient than parsing the baseURI string (especially because we have some services that use regional base URIs https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/regional_baseuri.go)

Reply:
`auth.CloudEnvironment()` will return the cloud name